eras/conway/impl/src/Cardano/Ledger/Conway/PParams.hs

```diff
@@ -154,6 +166,34 @@ class BabbageEraPParams era => ConwayEraPParams era where
   hkdMinFeeRefScriptCostPerByteL ::
     HKDFunctor f => Lens' (PParamsHKD f era) (HKD f NonNegativeInterval)
 
+instance ConwayEraPParams era => ToPlutusData (PParamsUpdate era) where
+  toPlutusData ppu =
+    P.Map (foldr' accum ([] :: [(P.Data, P.Data)]) (eraPParams @era))
+    where
+      accum PParam {ppUpdateTag, ppUpdateLens} acc =
+        let mbData = do
+              tag <- ppUpdateTag
+              t <- strictMaybeToMaybe $ ppu ^. ppUpdateLens
+              pure (P.I (toInteger @Word tag), toPlutusData t)
+         in maybe acc (: acc) mbData
```

Comment:
Slight simplification:
```suggestion
  toPlutusData ppu = P.Map (mapMaybe ppToData (eraPParams @era))
    where
      ppToData :: PParam era -> Maybe (P.Data, P.Data)
      ppToData PParam {ppUpdateTag, ppUpdateLens} = do
        tag <- ppUpdateTag
        t <- strictMaybeToMaybe $ ppu ^. ppUpdateLens
        pure (P.I (toInteger @Word tag), toPlutusData t)
```

Reply:
I hadn't realized I wrote a (probably worse) implementation of `mapMaybe` here. Thanks!